src/game/Pet.cpp

```diff
@@ -1101,148 +1076,165 @@ bool Pet::InitStatsForLevel(uint32 petlevel, Unit* owner)
         SetObjectScale(scale);
         UpdateModelData();
     }
-    m_bonusdamage = 0;
+
+    if (owner->GetTypeId() == TYPEID_PLAYER)
+        m_bonusdamage = 0;
 
     int32 createResistance[MAX_SPELL_SCHOOL] = {0, 0, 0, 0, 0, 0, 0};
 
     if (getPetType() != HUNTER_PET)
     {
-        createResistance[SPELL_SCHOOL_HOLY]   = cinfo->ResistanceHoly;
-        createResistance[SPELL_SCHOOL_FIRE]   = cinfo->ResistanceFire;
-        createResistance[SPELL_SCHOOL_NATURE] = cinfo->ResistanceNature;
-        createResistance[SPELL_SCHOOL_FROST]  = cinfo->ResistanceFrost;
-        createResistance[SPELL_SCHOOL_SHADOW] = cinfo->ResistanceShadow;
-        createResistance[SPELL_SCHOOL_ARCANE] = cinfo->ResistanceArcane;
+        createResistance[SPELL_SCHOOL_HOLY]   = cInfo->ResistanceHoly;
+        createResistance[SPELL_SCHOOL_FIRE]   = cInfo->ResistanceFire;
+        createResistance[SPELL_SCHOOL_NATURE] = cInfo->ResistanceNature;
+        createResistance[SPELL_SCHOOL_FROST]  = cInfo->ResistanceFrost;
+        createResistance[SPELL_SCHOOL_SHADOW] = cInfo->ResistanceShadow;
+        createResistance[SPELL_SCHOOL_ARCANE] = cInfo->ResistanceArcane;
     }
 
+    for (int i = SPELL_SCHOOL_HOLY; i < MAX_SPELL_SCHOOL; ++i)
+        SetModifierValue(UnitMods(UNIT_MOD_RESISTANCE_START + i), BASE_VALUE, float(createResistance[i]));
+
     switch (getPetType())
     {
         case SUMMON_PET:
         {
-            if (owner->GetTypeId() == TYPEID_PLAYER)
-            {
-                switch (owner->getClass())
-                {
-                    case CLASS_WARLOCK:
-                    {
-                        // the damage bonus used for pets is either fire or shadow damage, whatever is higher
-                        uint32 fire  = owner->GetUInt32Value(PLAYER_FIELD_MOD_DAMAGE_DONE_POS + SPELL_SCHOOL_FIRE);
-                        uint32 shadow = owner->GetUInt32Value(PLAYER_FIELD_MOD_DAMAGE_DONE_POS + SPELL_SCHOOL_SHADOW);
-                        uint32 val  = (fire > shadow) ? fire : shadow;
-
-                        SetBonusDamage(int32(val * 0.15f));
-                        // bonusAP += val * 0.57;
-                        break;
-                    }
-                    case CLASS_MAGE:
```

Comment:
What about the warlock pets?

> [Patch 2.0.3](http://wowwiki.wikia.com/wiki/Patch_2.0.3_(undocumented_changes))
> Warlock pet spells now scale with gear and talents (instead of only melee attack power).